src/code/qualification/plotQualificationPKRatio.m

```
@@ -43,93 +32,190 @@
-        pp=plot(X(i).*Xfactor, PKpred(i)/PKobs(i), 'Linewidth',1);
-        setCurveOptions(pp, CurveOptions);
-        
-    else
-        ME = MException('PKRatio:notFoundInPath', ...
-            'PKParameter %s not found using getPKParametersForConcentration', PKParameter);
-        throw(ME);
+    % Get PK parameters from the curves
+    % For simulation
+    allPKpred=getPKParametersForConcentration(SimTime, Pred, 'Dose', drugmass);
+    
+    % For Observations
+    allPKobs=getPKParametersForConcentration(ObsTime, Obs, 'Dose', drugmass);
+    
+    for k=1:length(PKParameter)
+        % Get the PK parameters requested in PKParameter
+        if strcmpi(PKParameter{k}, 'AUC')
+            PKpredField{k}= 'AUC_last';
+        elseif strcmpi(PKParameter{k}, 'CMAX')
+            PKpredField{k}= 'cMax';
+        else
+            PKpredField{k}=PKParameter{k};
+        end
+        if isfield(allPKpred, PKpredField{k}) && isfield(allPKobs, PKpredField{k})
+            
+            % Get the observation PK
+            Result.obsPK(i, k) = getfield(allPKobs, PKpredField{k});
+            % Get the predicted PK
+            Result.predPK(i, k) = getfield(allPKpred, PKpredField{k});
+            % Get the Ration
+            Result.RatioPK(i, k) = Result.predPK(i, k)./Result.obsPK(i, k);
+            
+        else
+            ME = MException('PKRatio:notFoundInField', ...
+                'Requested PK Parameter "%s" not found in parameters extracted using getPKParametersForConcentration', PKParameter{k});
+            throw(ME);
+        end
+    end
+end
+
+%--------------------------------------
+% Plot Section
+% Get X parameter:
+for k=1:length(AxesOptions)
+    if strcmp(AxesOptions(k).Type, 'X')
+        Xparam = AxesOptions(k).Dimension;
+        % Field is saved as AGE, similar process can be
+        % implmented for different X parameters
+        if strcmpi(Xparam, 'AGE')
+            Xparam = 'AGE';
+        end

```

@pchelle Here I would throw an exception if it's NOT `Age `(just in case if we extend it in the future)